### PR TITLE
Update issue_index to finish migration

### LINKED ISSUE #16683: [Bug] issue_repo_index not set for migrated repos
gitea version: v1.15.0-rc1 or newer

introduced with #15599

reproduce:
1. migrate any repo with issues/pulls
2. try to create a new pull or issue

-> `newIssue: Error 1062: Duplicate entry '424-1' for key 'UQE_tmp_recreate__issue_repo_index'`

### PR DESCRIPTION
**Pull request by 6543**
fix #16683

### CODE REVIEW

**Comment by 6543** [comment]
@lunny do you think it can be done bit more elegant?

File: modules/migrations/gitea_uploader.go [hunk 1]
```diff
@@ -871,6 +871,20 @@ func (g *GiteaLocalUploader) Finish() error {
...
+	max, err := models.GetMaxIssueIndex(g.repo.ID)
+	if err != nil {
+		return err
+	}
+
+	if _, err := models.GetNextResourceIndex("issue_index", g.repo.ID); err != nil {
+		return err
+	}
+
+	if err := models.UpdateResourceIndex("issue_index", g.repo.ID, max); err != nil {
```
Comment (zeripath) [inline modules/migrations/gitea_uploader.go]: Shouldn't this all be done in one transaction?

**Review by zeripath — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
Please send backport to v1.15

**Comment by 6543** [comment]
-> #16687

**Pull request merged**